[BUG] CI premerge runs tests against nightly snapshot artifact

While digging into CI failures for #11375, I noticed that the premerge CI build is downloading the nightly snapshot of the plugin during the build.  That means it's not testing against all of the changes in the PR, at least for that portion of the CI job.  It appears to be related to a `mvn test` which ends up locating the dependency artifacts in the nightly snapshot repository rather than using built classes from dependency projects.

I verified that a `mvn test` with a clean m2 local repo fails with:
```
[ERROR] Failed to execute goal on project rapids-4-spark-sql_2.12: Could not resolve dependencies for project com.nvidia:rapids-4-spark-sql_2.12:jar:24.10.0-SNAPSHOT: Could not find artifact com.nvidia:rapids-4-spark-sql-plugin-api_2.12:jar:spark320:24.10.0-SNAPSHOT in snapshots-repo (https://oss.sonatype.org/content/repositories/snapshots) -> [Help 1]
```
which implies that if it could have found the snapshot artifacts, it would have used them rather than the classes that were just built.  Using a `mvn package` instead of `mvn test` works around the issue.

--- a/jenkins/spark-premerge-build.sh
+++ b/jenkins/spark-premerge-build.sh
@@ -70,7 +70,7 @@ mvn_verify() {
     for version in "${SPARK_SHIM_VERSIONS_PREMERGE_UTF8[@]}"
     do
         echo "Spark version (regex): $version"
-        env -u SPARK_HOME LC_ALL="en_US.UTF-8" $MVN_CMD $MVN_URM_MIRROR -Dbuildver=$version test $MVN_BUILD_ARGS \
+        env -u SPARK_HOME LC_ALL="en_US.UTF-8" $MVN_CMD $MVN_URM_MIRROR -Dbuildver=$version package $MVN_BUILD_ARGS \
           -Dpytest.TEST_TAGS='' \
           -DwildcardSuites=com.nvidia.spark.rapids.ConditionalsSuite,com.nvidia.spark.rapids.RegularExpressionSuite,com.nvidia.spark.rapids.RegularExpressionTranspilerSuite
     done

